### PR TITLE
Add doc for thanos rule stateless mode

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

This pr adds the missing document for Thanos Ruler stateless mode.

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

### CODE REVIEW

File: docs/components/rule.md [hunk 1]
```diff
@@ -202,6 +204,35 @@ In case of Ruler in HA you need to make sure you have the following labelling se
...
+    --objstore.config-file      "bucket.yml" \
+    --label                     'monitor_cluster="cluster1"' \
+    --label                     'replica="A"' \
+    --remote-write.config-file  'rw-config.yaml'
+```
+
+The remote write config file is exactly the same as the [Prometheus remote write config format](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write).
+
+**NOTE:**
+1. Remote write metadata config is not supported in this mode and will be ignored if provided in the remote write configuration.
```
Comment (GiedriusS) [inline docs/components/rule.md:233]: What's "remote write metadata config"? :thinking:

File: docs/components/rule.md [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
 The `thanos rule` command evaluates Prometheus recording and alerting rules against chosen query API via repeated `--query` (or FileSD via `--query.sd`). If more than one query is passed, round robin balancing is performed.
 
-Rule results are written back to disk in the Prometheus 2.0 storage format. Rule nodes at the same time participate in the system as source store nodes, which means that they expose StoreAPI and upload their generated TSDB blocks to an object store.
+By default, rule evaluation results are written back to disk in the Prometheus 2.0 storage format. Rule nodes at the same time participate in the system as source store nodes, which means that they expose StoreAPI and upload their generated TSDB blocks to an object store.
+
+Rule also has a stateless mode which sends rule evaluation results to some remote storages via remote write for better scalability. This way, rule nodes only work as a data producer and the remote receive nodes work as source store nodes.
```
Comment (GiedriusS) [inline docs/components/rule.md:11]: Maybe let's clarify the difference even more:

```suggestion
Rule also has a stateless mode which sends rule evaluation results to some remote storages via remote write for better scalability. This way, rule nodes only work as a data producer and the remote receive nodes work as source store nodes. It means that Thanos Rule in this mode does _not_ expose the StoreAPI.
```

File: docs/components/rule.md (hunk 1 above)
Comment (yeya24) [reply docs/components/rule.md:233]: image

Metadata config depends on scrape manager so it is not used.

File: docs/components/rule.md (hunk 2 above)
Comment (yeya24) [reply docs/components/rule.md]: Done

**Comment by hanjm** [comment]
Awsome, LGTM

File: docs/components/rule.md (hunk 1 above)
Comment (GiedriusS) [reply docs/components/rule.md]: Ahh, I see. Maybe we can then replace "remote write metadata config" with "remote write `metadata_config`" to emphasize that it is a key's name?

Comment (yeya24) [reply docs/components/rule.md]: Done

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**